### ISSUE TITLE
determine for which domains we failed to obtain a certificate for

### ISSUE THREAD

**Issue by marten-seemann** [caddyserver/certmagic#173]
First of all, a big thank you, @mholt, for this terrific library!

## What would you like to have changed?

When obtaining certificates for a number of domains, it would be nice to be able to know for which domains a certificate was successfully be obtained and for which domains it failed.

`cfg.ManageAsync` doesn't give error return values, but only
`cfg.ManageSync` returns as soon as the first request fails, therefore can't be used be used for that purpose.

## Why is this feature a useful, necessary, and/or important addition to this project?

I'm building a (lib)p2p node integration that uses TLS certificates to secure WebSocket connections. WebSocket support is optional, so if obtaining one (or a few, or even all) certificates fails, that's not the end of the world. Certificates are therefore obtained asynchronously.
The node will just disable support for failing domains and move on. It would therefore be nice to be able to learn which domains we successfully obtained a certificate for, and which ones it failed.

## What alternatives are there, or what are you doing in the meantime to work around the lack of this feature?

I'm assuming that `cfg.ManageSync(domains)` is more efficient than doing a
```go
for _, domain := range domains {
    cfg.ManageSync(domain)
}
```
If that's not the case, this issue is probably moot.

**Comment by francislavoie** [comment]
There's event hooks in certmagic, which I opened a PR to improve recently: https://github.com/caddyserver/certmagic/pull/150 this will probably do what you need!

**Comment by marten-seemann** [comment]
I assume I'd then call `ManageAsync` and listen for the `cert_obtained` event, right? Would that event also be emitted when obtain the cert fails? In #150, it doesn't look like there's an error field in that event.

**Comment by marten-seemann** [comment]
Actually, I think I don't need it. There'd be no difference between "this domain is still waiting for the cert" and "we (permanently?) failed to obtain a cert for this domain". In both cases, I know I can't use that domain yet.

**Comment by mholt** [comment]
Should be possible now that we've merged more detailed certificate events in #150. In case you still want to know :man_shrugging:
